server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -491,7 +513,7 @@ private void updateCleanupKeyToCountMapOnCacheInsertion(CleanupKey cleanupKey) {
         }
 
         private void updateCleanupKeyToCountMapOnCacheEviction(CleanupKey cleanupKey) {
-            if (stalenessThreshold == 0.0 || cleanupKey.entity == null) {
+            if (cleanupKey.entity == null) {
```

Comment:
Looks like we have a problem here. If any evicted key from cache was referring to older shard, cleanup.entity would be null(due to current logic) and we may end up not decrementing staleKeyCount correctly. Probably verify via IT as well.
Might require a deeper fix in existing code, lets discuss on this.